### PR TITLE
[FEATURE] ID/PK flag `return_unexpected_index_query` that allows `QUERY` output to be suppressed

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
- Adds `return_unexpected_index_query` flag to `result_format`. 
- By default, `unexpected_index_query` is returned by `ExecutionEngines` that support it (currently on `SqlAlchemyExecutionEngine`), only when `result_format` is set to `COMPLETE`. 
- setting `return_unexpected_index_query` to `False` will suppress the `unexpected_index_query` even when `result_format` is set to `COMPLETE`
-  Added tests at `tests/checkpoint/test_checkpoint_result_format.py`-level

### Can you give some examples? 
```python 
        "result_format": {
            "result_format": "COMPLETE",
            "unexpected_index_column_names": ["pk_1"],
            "return_unexpected_index_query": True,
        }
```
* **Will** have `unexpected_index_query`, since `return_unexpected_index_query` is explicitly set to `True`

```python 
        "result_format": {
            "result_format": "COMPLETE",
            "unexpected_index_column_names": ["pk_1"],
            "return_unexpected_index_query": False,
        }
```
* **Will not** have `unexpected_index_query`, since `return_unexpected_index_query` is explicitly set to `False`

```python 
        "result_format": {
            "result_format": "COMPLETE",
            "unexpected_index_column_names": ["pk_1"],
        }
```
* **Will** have `unexpected_index_query`, since it is returned by default if result format is set to `COMPLETE`

```python 
        "result_format": {
            "result_format": "SUMMARY",
            "unexpected_index_column_names": ["pk_1"],
        }
```
* **Will not** have `unexpected_index_query`, since result format is set to `SUMMARY`

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/checkpoint/test_checkpoint_result_format.py [hunk 1]
```diff
@@ -282,7 +411,7 @@ def test_sql_result_format_not_in_checkpoint_passed_into_run_checkpoint_one_expe
         expectations_list=[expectation_config_expect_column_values_to_be_in_set],
     )
     result_format: dict = {
-        "result_format": "SUMMARY",
+        "result_format": "COMPLETE",
```
Comment (Shinnnyshinshin) [inline tests/checkpoint/test_checkpoint_result_format.py:415]: This was a bug. Test was for `COMPLETE` , but we were passing in `SUMMARY`.

File: tests/checkpoint/test_checkpoint_result_format.py [hunk 2]
```diff
@@ -742,6 +887,7 @@ def test_sql_result_format_in_checkpoint_pk_defined_one_expectation_summary_outp
         "partial_unexpected_index_list"
     ]
     assert first_result_partial_list == expected_unexpected_indices_output
+    assert evrs[0]["results"][0]["result"].get("unexpected_index_query") is None
```
Comment (Shinnnyshinshin) [inline tests/checkpoint/test_checkpoint_result_format.py:891]: `SUMMARY` output does not have `unexpected_index_query`

File: great_expectations/expectations/metrics/map_metric_provider.py [hunk 3]
```diff
@@ -2411,6 +2411,13 @@ def _sqlalchemy_map_condition_query(
     domain_kwargs: dict = dict(**compute_domain_kwargs, **accessor_domain_kwargs)
     result_format: dict = metric_value_kwargs["result_format"]
 
+    # We will not return map_condition_query if return_unexpected_index_query = False
+    return_unexpected_index_query: Union[bool, None] = result_format.get(
```
Comment (rdodev) [inline great_expectations/expectations/metrics/map_metric_provider.py]: general question: Why do we need this Union type here? Woudn't python's truthiness take into account whether something is `False` or `None` => falsey and wouldn't it have then same effect?

Comment (Shinnnyshinshin) [reply great_expectations/expectations/metrics/map_metric_provider.py]: I think you're totally right :) Thanks for catching this @rdodev

**Review by rdodev — APPROVED** [review]
LGTM. :shipit:

**Pull request merged**